app/views/elastic_cron_helpers.py

```diff
@@ -19,83 +31,38 @@ def cron_rebuild_events_elasticsearch():
-    """
-    def __init__(self, high, set_name):
-        self.current = 1
-        self.high = high
-        self.set_name = set_name
 
-    def __iter__(self):
-        return self
+    es_store.indices.delete(SearchableEvent.meta.index)
+    SearchableEvent.init()  # Create ES index
 
-    def __next__(self):
-        if self.current >= self.high:
-            raise StopIteration
-        else:
-            self.current += 1
-            event_id = redis_store.spop(self.set_name)
-            cur = conn.cursor()
-            cur.execute(
-                "SELECT id, name, description, searchable_location_name, organizer_name, organizer_description FROM events WHERE id = %s;",
-                (event_id,))
-            event_ = cur.fetchone()
-            return event_
+    for event in session.query(Event):
+        searchable = SearchableEvent()
+        searchable.from_event(event)
+        searchable.save()
 
 
 def sync_events_elasticsearch():
     # Sync update and inserts
-    index_count = redis_store.scard('event_index')
-    index_event_data = ({'_type': 'event',
-                         '_index': 'events',
-                         '_id': event_[0],
-                         'name': event_[1],
-                         'description': event_[2] or None,
-                         'searchable_location_name': event_[3] or None,
-                         'organizer_name': event_[4] or None,
-                         'organizer_description': event_[5] or None}
-                        for event_ in EventIterator(index_count, 'event_index'))
-    try:
-        helpers.bulk(es_store, index_event_data)
-    except Exception as e:
-        print(e)
+    updated, todo = 0, redis_store.scard('event_index')
+
+    while updated < todo:
+        updated += 1
+        event_id = redis_store.spop('event_index')
+        event = session.query(Event).filter(id=event_id).one()
+        searchable = SearchableEvent()
+        searchable.from_event(event)
+        searchable.save()
 
     # sync both soft and hard deletes
-    del_count = redis_store.scard('event_delete')
-    del_event_data = ({'_type': 'event',
-                       '_index': 'events',
-                       '_id': event_[0],
-                       'name': event_[1],
-                       'description': event_[2] or None,
-                       'searchable_location_name': event_[3] or None,
-                       'organizer_name': event_[4] or None,
-                       'organizer_description': event_[5] or None}
-                      for event_ in EventIterator(del_count, 'event_delete'))
-    try:
-        helpers.bulk(es_store, del_event_data)
-    except Exception as e:
-        print(e)
+    deleted, todo = 0, redis_store.scard('event_delete')
+
+    while deleted < todo:
+        deleted += 1
+        event_id = redis_store.spop('event_delete')
+        searchable = SearchableEvent()
+        searchable.meta.id = event_id
+
+        if event_id:  # For safety
```

Comment:
What is the need of this `if` block?

Reply:
On create, the id is taken from the database and therefore can't be `None`. On delete however, it is possible that redis is out of sync so I double check the id here. It's not completely necessary though